File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.java
import java.util.Scanner;

public class Main {
    static boolean[] arr;
    static int switchSize, n, gender, number;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        switchSize = sc.nextInt();
        arr = new boolean[switchSize];
        for (int i = 0; i < switchSize; i++) {
            int a;
            a = sc.nextInt();
            if (a == 1) arr[i] = true;
            else arr[i] = false;
        }
        n=sc.nextInt();
        while(n-->0){
            gender=sc.nextInt();
            number=sc.nextInt();
            if(gender==1){
                for(int i=number-1;i<switchSize;i+=number){
                    arr[i]=!arr[i];
                }
            }
            else{
                number--;
                arr[number]=!arr[number];
                for(int i=1;;i++){
                    if (number - i < 0 || switchSize <= number + i||
                            (arr[number-i]!=arr[number+i]))break;
                    arr[number - i] = !arr[number - i];
                    arr[number + i] = !arr[number + i];
                }
            }
        }
        for(int i=1;i<=switchSize;i++){
            if(arr[i-1]) System.out.printf("1 ");
            else System.out.printf("0 ");
            if(i%20==0) System.out.println();
        }
    }
}

File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.cc
#include<iostream>
using namespace std;
bool arr[100];
int switchSize, n, gender, number;
int main() {
	cin >> switchSize;
	for (int i = 0; i < switchSize; i++) {
		int a;
		cin >> a;
		if (a)arr[i] = 1;
		else arr[i] = 0;
	}
	cin >> n;
	while (n--) {
		cin >> gender >> number;
		if (gender == 1) {

			for (int i = number - 1; i < switchSize; i += number) {
				arr[i] = !arr[i];
			}
		}
		else {
			number--;
			arr[number] = !arr[number];
			for (int i = 1;; i++) {
				if (number - i < 0 || switchSize <= number + i||
					(arr[number-i]!=arr[number+i]))break;
				arr[number - i] = !arr[number - i];
				arr[number + i] = !arr[number + i];
				
			}
		}
	}
	for (int i = 1; i <= switchSize; i++) {
		if (arr[i-1])cout << 1 << " ";
		else cout << 0 << " ";
		if (i % 20 == 0)cout << "\n";

		

	}
	return 0;
}